lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -130,6 +130,27 @@ def set_3d_properties(self, zs=0, zdir='z'):
         self._verts3d = juggle_axes(xs, ys, zs, zdir)
         self.stale = True
 
+    def set_data_3d(self, *args):
+        """
+        Set the x, y and z data
+
+        ACCEPTS: 3D array (rows are x, y, z) or three 1D arrays
+        """
+        if len(args) == 1:
+            x, y, z = args[0]
+        else:
+            x, y, z = args
+
+        self.set_xdata(x)
+        self.set_ydata(y)
```

Comment:
This is incorrect. `set_xdata()` and `set_ydata()` are for the 2D projected values. `self._verts3d` is just a tuple of the 3 1D arrays.

Reply:
Ah, fair enough.  Using this method is very suggestive due to

https://github.com/matplotlib/matplotlib/blob/37978a8a701530acc824500c7d4a09cad1d93d7c/lib/mpl_toolkits/mplot3d/art3d.py#L117-L129

but on a bit closer inspection it seems that these methods are used in the 2D class -> 3D class upcasting, not as a standard part of the setup.